pkg/api/db.go

```diff
@@ -203,7 +205,7 @@ func DeleteVolumeSnapshotDBEntry(ctx *c.Context, in *model.VolumeSnapshotSpec) e
 //Just modify the state of the volume to be deleted in the DB, the real deletion in another thread
 func DeleteVolumeDBEntry(ctx *c.Context, in *model.VolumeSpec) error {
 	invalidStatus := []string{model.VolumeAvailable, model.VolumeError,
-		model.VolumeErrorDeleting, model.VolumeErrorExtending}
+		model.VolumeErrorDeleting, model.VolumeErrorExtending, model.VolumeDeleted}
```

Comment:
Add volumeCreating and volumeDeleting statuses and remove volumeErrorDeleting status.